src/ilabs_streamsync/streamsync.py

```diff
@@ -9,39 +20,104 @@ class StreamSync:
     """
 
     def __init__(self, reference_object, pulse_channel):
-        self.ref_stream = reference_object.get_chan(pulse_channel)
-        self.sfreq = reference_object.info["sfreq"]  # Hz
-        self.streams = []
+        """Initialize StreamSync object with 'Raw' MEG associated with it."""
+        # self.ref_stream = reference_object.get_chan(pulse_channel)
+        self.ref_stream = None
+        # self.sfreq = reference_object.info["sfreq"]  # Hz
+        self.sfreq = 0
```

Comment:
Initially I was leaving this unimplemented such that other functionality of the class could be tested before adding this. I could similarly change the code to your suggestion and leave the get_data() function as a placeholder, or I could begin with this if you find it important to have the workflow in this order.

Reply:
My thinking is that you can't meaningfully test the other parts until this is working, because the design is that the *other* time domains will be shifted/stretched/resampled to match the *MEG data* time domain.  In other words, if you can't load the reference object, then you can't sync anything else to it (and so you can't test if your syncing works).

Reply:
That's quite fair, I suppose I was initially doing it this way since the alignment process is also capable of aligning two individual cameras, but I agree that this is pointless without an absolute baseline. I'll make this change first and stash some of the other changes for now.